wcfsetup/install/files/lib/system/worker/SitemapRebuildWorker.class.php

```diff
@@ -421,4 +429,25 @@ private function changeUserToGuest() {
 	private function changeToActualUser() {
 		WCF::getSession()->changeUser($this->actualUser, true);
 	}
+	
+	/**
+	 * Reads the columns changed by the user for this sitemap object from the registry and modifies the object accordingly.
+	 * 
+	 * @param       ObjectType      $object
+	 * @since       5.3
+	 */
+	public static function prepareSitemapObject(ObjectType &$object) {
+		$sitemapData = RegistryHandler::getInstance()->get('com.woltlab.wcf', self::REGISTRY_PREFIX . $object->objectType);
+		
+		if ($sitemapData !== null) {
+			$sitemapData = @unserialize($sitemapData);
+			
+			if (is_array($sitemapData)) {
+				$object->priority = $sitemapData['priority'];
+				$object->changeFreq = $sitemapData['changeFreq'];
+				$object->rebuildTime = $sitemapData['rebuildTime'];
+				$object->isDisabled = $sitemapData['isDisabled'];
```

Comment:
This is unsafe, because it adds new public properties to the ObjectType object, instead of syncing the internal `data` member.

Reply:
Yes, but the data attribute is `protected`. I could of course create a new object with the data, but that would also be _wrong_ somehow. I think the current solution is sufficient, especially because you only access the data via the Magic-Getter.

Reply:
I don't really see an issue here, the implicit properties are not persistent and are valid for the lifetime of the request only. Also, these values are in fact correct and thus will not break any code that for whatever reason accesses them.